src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserServiceTest.java

```diff
@@ -2,296 +2,391 @@
+        UserDTO userDTO = new UserDTO();
+        userDTO.setUsername(""); // Празно потребителско име
 
-        when(userRepository.findById(userId)).thenReturn(Optional.of(existingUser));
+        when(userRepository.findByIdAndDeletedFalse(userId)).thenReturn(Optional.of(existingUser));
 
-        assertThrows(ApiRequestException.class, () -> userService.updateUserWithoutPassword(userId, userDto));
-
-        verify(userRepository, times(1)).findById(userId);
-        verify(userRepository, never()).save(any());
-        verify(userMapper, never()).toDto(any());
+        // Act and Assert
+        assertThrows(jakarta.validation.ValidationException.class, () -> {
+            userService.updateUser(userId, userDTO);
+        });
     }
 
     @Test
-    void testDeleteUser_ExistingUser() {
+    void testUpdateUser_InvalidUser_FullNameBlank() {
+        // Arrange
         Long userId = 1L;
-
         User existingUser = new User();
         existingUser.setId(userId);
-        existingUser.setUsername("john_doe");
+        existingUser.setUsername("user1");
+        existingUser.setFullName("User 1");
+        existingUser.setEmail("user1@example.com");
+        existingUser.setRole(Role.ADMIN);
 
-        when(userRepository.findById(userId)).thenReturn(Optional.of(existingUser));
+        UserDTO userDTO = new UserDTO();
+        userDTO.setUsername("user1");
+        userDTO.setFullName(""); // Празно потребителско име
 
-        userService.deleteUser(userId);
+        when(userRepository.findByIdAndDeletedFalse(userId)).thenReturn(Optional.of(existingUser));
 
-        verify(userRepository, times(1)).findById(userId);
-        verify(userRepository, times(1)).delete(existingUser);
+        // Act and Assert
+        assertThrows(jakarta.validation.ValidationException.class, () -> {
+            userService.updateUser(userId, userDTO);
+        });
     }
 
     @Test
-    void testDeleteUser_NonExistingUser() {
+    void testUpdateUser_InvalidUser_EmailBlank() {
+        // Arrange
         Long userId = 1L;
+        User existingUser = new User();
+        existingUser.setId(userId);
+        existingUser.setUsername("user1");
+        existingUser.setFullName("User 1");
+        existingUser.setPassword("passs");
+        existingUser.setEmail("");
+        existingUser.setRole(Role.ADMIN);
+
+        UserDTO userDTO = new UserDTO();
+        userDTO.setUsername("user1");
+        userDTO.setFullName("user user 1 ");
+        userDTO.setEmail(""); // Празно потребителско име
+        userDTO.setRole(Role.ADMIN);
+        when(userRepository.findByIdAndDeletedFalse(userId)).thenReturn(Optional.of(existingUser));
+
+        // Act and Assert
+        assertThrows(jakarta.validation.ValidationException.class, () -> {
+            userService.updateUser(userId, userDTO);
+        });
+    }
+
 
-        when(userRepository.findById(userId)).thenReturn(Optional.empty());
+    @Test
+    void testCreateUser_UsernameBlank() {
```

Comment:
We can also have a test method for username with less that 4 symbols (the minimum required symbols) and it should not create the user.

Reply:
added try test for create with short name